allennlp/data/tokenizers/word_filter.py

```diff
@@ -33,38 +33,71 @@ def filter_words(self, words: List[Token]) -> List[Token]:
         return words
 
 
+@WordFilter.register('regex')
+class RegexFilter(WordFilter):
+    """
+    Uses a list of regexes to filter
+    """
+    def __init__(self,
+                 patterns: List[str]) -> None:
+        self._patterns = patterns
+        self._joined_pattern = "|".join(self._patterns)
+
+    @overrides
+    def filter_words(self, words: List[Token]) -> List[Token]:
+        stopwords = [word for word in words
+                      if re.match(pattern=self._joined_pattern, string=word.text)]
+        return [word for word in words if word not in stopwords]
```

Comment:
You're iterating over things twice here.  It's probably better to just have a single loop over the words, check for `re.match`, and append to a list.